### PR TITLE
Context Plane

### PR DESCRIPTION
**Pull request by TaylorBeeston**
- :sparkles: Add Context Plane to @learncard/core
- :sparkles: Add context loading support to didkit plugin
- :sparkles: Create Dynamic Loader plugin
- :sparkles: Add Dynamic Loader plugin to init
- :sparkles: Allow loading remote contexts in CLI
- :bookmark: Changesets

# Overview

#### 📚 What is the context and goal of this PR?
Right now, it's really hard to add missing contexts. We have to add them to SSI and recompile didkit, 
host a new wasm binary, and push new versions to npm. This is especially rough when needing to resolve
odd, bespoke contexts, such as the sample CHAPI contexts. This PR is intended to deal with that by
adding both a means to easily add custom contexts, as well as a mechanism for dynamically resolving
contexts!

#### 🥴 TL; RL:
Adds Context Plane, Dynamic Loader plugin, and updates Didkit plugin to make use of the context plane!

#### 💡 Feature Breakdown (screenshots & videos encouraged!)
TODO: I will detail this better in the morning!

#### 🛠 Important tradeoffs made:
It is kind of scary to make it _so easy_ to dynamically resolve contexts, so I wonder if maybe we
should instead force consumers to actually add the plugin after the fact rather than just pass in
a simple flag 🤔

#### 🔍 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore (refactor, documentation update, etc)

#### 💳 Does This Create Any New Technical Debt? ( If yes, please describe and [add JIRA TODOs](https://welibrary.atlassian.net/jira/software/projects/WE/boards/2) )
- [x] No
- [ ] Yes

# Testing

#### 🔬 How Can Someone QA This?

Go to VC Playground, issue the 'Safe Chef' credential, then copy it into the CLI and attempt to verify
it! The CLI is now by default dynamically resolving contexts, and I can 100% confirm that the safe chef
context is _not_ in SSI, so if it verifies, that means it sent the request and got it!

You can double check this in the CLI by making a new wallet without dynamic resolution (`let test = await initLearnCard()`),
and then attempt to verify the same credential with it. You should see an error about context expansion!

#### 🧪 Code Coverage

# Documentation

#### 📜 Gitbook

https://app.gitbook.com/o/6uDv1QDlxaaZC7i8EaGb/s/FXvEJ9j3Vf3FW5Nc557n/~/changes/75/learn-card-sdk/learncard-core/control-planes/context

#### 📊 Storybook

# ✅ PR Checklist
- [ ] Related to a Jira issue ([create one if not](https://welibrary.atlassian.net/jira/software/projects/WE/boards/2))
- [x] My code follows **style guidelines** (eslint / prettier)
- [x] I have **manually tested** common end-2-end cases
- [x] I have **reviewed** my code
- [x] I have **commented** my code, particularly where ambiguous
- [x] New and existing **unit tests pass** locally with my changes
- [x] I have made corresponding changes to **gitbook documentation**

### 🚀 Ready to squash-and-merge?:
- [x] Code is backwards compatible
- [x] There is **not** a "Do Not Merge" label on this PR
- [x] I have thoughtfully considered the security implications of this change.
- [x] This change does not expose new public facing endpoints that do not have authentication

### CODE REVIEW

**Review by Custard7 — COMMENTED** [review]
Hey @TaylorBeeston! Was trying to test, and seems like the nx deps need to be updated.

When I run `bun run nx start cli --skip-nx-cache` I get errors about the expiration-plugin

I left some inline comments, and in the learn-card-init project.json, I think you need to add dynamic-loader-plugin to the nx dependencies too!

Although, I was still running into some build funkiness

File: packages/plugins/dynamic-loader/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+    "name": "@learncard/dynamic-loader-plugin",
+    "version": "0.0.0",
```
[the remark this answers is not in the document]
Comment (TaylorBeeston) [reply packages/plugins/dynamic-loader/package.json:3]: Changesets will set this on when it releases!

**Comment by TaylorBeeston** [comment]
@Custard7 good catch! I updated the project.json, please try again! The only build issue I had was because I forgot to `pnpm i`. After that, everythign worked!

**Review by Custard7 — APPROVED** [review]
Yup! LGTM 👍 

Everything worked this time ✅ 

Screen Shot 2023-09-20 at 6 21 47 PM

**Pull request merged**